src/lib/gundb/FeedStorage.js

```diff
@@ -156,6 +161,123 @@ export class FeedStorage {
     this._syncFeedCache()
   }
 
+  /**
+   * callback to use when we get a transaction that withdrawn our payment link
+   * @param {*} receipt
+   */
+  async handleOTPLUpdated(receipt: any): Promise<FeedEvent> {
+    //receipt received via websockets/polling need mutex to prevent race
+    //with enqueuing the initial TX data
+    const release = await this.feedMutex.lock()
+    try {
+      const data = this.getReceiveDataFromReceipt(receipt, this.wallet.account)
+      logger.debug('handleOTPLUpdated', { data, receipt })
+
+      //get our tx that created the payment link
+      //paymentId is new format, hash is in old beta format
+      const originalTXHash = await this.getTransactionHashByCode(data.hash || data.paymentId)
+      if (originalTXHash === undefined) {
```

Comment:
```suggestion
    // receipt received via websockets/polling need mutex to prevent race
    // with enqueuing the initial TX data
    const release = await this.feedMutex.lock()
    
    try {
      const data = this.getReceiveDataFromReceipt(receipt, this.wallet.account)
      
      logger.debug('handleOTPLUpdated', { data, receipt })

      // get our tx that created the payment link
      // paymentId is new format, hash is in old beta format
      const originalTXHash = await this.getTransactionHashByCode(data.hash || data.paymentId)
      
      if (originalTXHash === undefined) {
```